code/search/SearchVariantSubsites.php

```diff
@@ -83,9 +83,16 @@ public function extractManipulationWriteState(&$writes)
                 continue;
             }
 
+            if (isset($write['fields']['SiteTree:SubsiteID'])) {
```

Comment:
Unfortunately Solr acts as an extremely de-normalised store; For any filter (e.g. any given subsite, stage, etc combination) a filter will be created; This is that json-looking string in the indexed data. If a particular model is visible within that filter (such as golbal files) then there needs to be a separate record saved for that combination.